src/main/java/org/jabref/model/texparser/TexParserResult.java

```diff
@@ -0,0 +1,87 @@
+package org.jabref.model.texparser;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.StringJoiner;
+
+public class TexParserResult {
+
+    private final List<Path> fileList;
+    private final List<Path> nestedFiles;
+    private final Map<String, Set<Citation>> citations;
```

Comment:
You can also use a multimap, which might be a bit more convenient: https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/HashMultimap.html